test/lib/tokenFactory.test.js

```diff
@@ -0,0 +1,187 @@
+'use strict';
+
+const assert = require('chai').assert;
+const mockery = require('mockery');
+const sinon = require('sinon');
+
+sinon.assert.expose(assert, { prefix: '' });
+
+describe('Token Factory', () => {
+    const password = 'super_secure_password';
+    const name = 'mobile_token';
+    const description = 'a token for a mobile app';
+    const sealed = 'abcd';
+    const unsealed = 'efgh';
+    const userId = 6789;
+    const tokenId = 12345;
+    const tokenData = {
+        id: tokenId,
+        userId,
+        description,
+        name,
+        value: sealed,
+        lastUsed: null
+    };
+    let TokenFactory;
+    let datastore;
+    let ironMock;
+    let factory;
+    let Token;
+
+    before(() => {
+        mockery.enable({
+            useCleanCache: true,
+            warnOnUnregistered: false
+        });
+    });
+
+    beforeEach(() => {
+        datastore = {
+            save: sinon.stub(),
+            scan: sinon.stub(),
+            get: sinon.stub()
+        };
+        ironMock = {
+            seal: sinon.stub(),
+            unseal: sinon.stub(),
+            defaults: 'defaults'
+        };
+
+        mockery.registerMock('iron', ironMock);
+
+        // eslint-disable-next-line global-require
+        Token = require('../../lib/token');
+        // eslint-disable-next-line global-require
+        TokenFactory = require('../../lib/tokenFactory');
+
+        factory = new TokenFactory({ datastore, password });
+    });
+
+    afterEach(() => {
+        datastore = null;
```

Comment:
> FWIW, the setup and teardown of these tests is pretty expensive

`useCleanCache` is a very expensive setup operation, but necessary.

As for teardown, it's a combination of `useCleanCache` and `.resetCache`. The `afterEach` is pretty redundant since `.deregisterAll` removes all the registered mocks from the cache, and then `.resetCache` just dumps it all and primes it again.